tests/devices/experimental/test_default_qubit_2.py

```diff
@@ -247,6 +247,226 @@ def test_tf_results_and_backprop(self):
         assert qml.math.allclose(grad1[0], -tf.sin(phi))
 
 
+class TestSampleMeasurements:
+    """A copy of the `qubit.simulate` tests, but using the device"""
+
+    def test_single_expval(self):
+        """Test a simple circuit with a single expval measurement"""
+        x = np.array(0.732)
+        qs = qml.tape.QuantumScript([qml.RY(x, wires=0)], [qml.expval(qml.PauliZ(0))], shots=10000)
+
+        dev = DefaultQubit2()
+        result = dev.execute(qs)
+
+        assert isinstance(result, np.ndarray)
+        assert result.shape == ()
+        assert np.allclose(result, np.cos(x), atol=0.1)
+
+    def test_single_probs(self):
+        """Test a simple circuit with a single prob measurement"""
+        x = np.array(0.732)
+        qs = qml.tape.QuantumScript([qml.RY(x, wires=0)], [qml.probs(wires=0)], shots=10000)
+
+        dev = DefaultQubit2()
+        result = dev.execute(qs)
+
+        assert isinstance(result, np.ndarray)
+        assert result.shape == (2,)
+        assert np.allclose(result, [np.cos(x / 2) ** 2, np.sin(x / 2) ** 2], atol=0.1)
+
+    def test_single_sample(self):
+        """Test a simple circuit with a single sample measurement"""
+        x = np.array(0.732)
+        qs = qml.tape.QuantumScript([qml.RY(x, wires=0)], [qml.sample(wires=range(2))], shots=10000)
+
+        dev = DefaultQubit2()
+        result = dev.execute(qs)
+
+        assert isinstance(result, np.ndarray)
+        assert result.shape == (10000, 2)
+        assert np.allclose(
+            np.sum(result, axis=0).astype(np.float32) / 10000, [np.sin(x / 2) ** 2, 0], atol=0.1
+        )
+
+    def test_multi_measurements(self):
+        """Test a simple circuit containing multiple measurements"""
+        x, y = np.array(0.732), np.array(0.488)
+        qs = qml.tape.QuantumScript(
+            [qml.RX(x, wires=0), qml.CNOT(wires=[0, 1]), qml.RY(y, wires=1)],
+            [qml.expval(qml.PauliZ(0)), qml.probs(wires=range(2)), qml.sample(wires=range(2))],
```

Comment:
We should test with non-commuting measurements.
```suggestion
            [qml.expval(qml.PauliX(0)), qml.probs(wires=range(2)), qml.sample(wires=range(2))],
```

Something that we couldn't do before, but works fine now.